test/test_filters.rb

```diff
@@ -958,6 +958,14 @@ def to_liquid
         results = @filter.where(SelectDummy.new, "obj", "1 == 1")
         assert_equal [], results
       end
+
+      should "gracefully handle invalid property type" do
+        hash = {
+          "members" => { "name" => %w(John Jane Jimmy) },
+          "roles"   => %w(Admin Recruiter Manager),
+        }
+        assert_raises(TypeError) { @filter.where(hash, "name", "Jimmy") }
```

Comment:
Hm, I think we want to add an assertion on the message here; this would pass without your change I think, so there's no way to know if we broke it.

Reply:
Added an assertion.